version: '3.8'
services:
  db_tests:
    image: postgres:15.1-alpine
    volumes:
      - db_tests:/var/lib/postgresql/data/
    env_file:
      - ./.env
    environment:
      - DB_HOST=db_tests
    ports:
      - "5431:5432"
    networks:
      - tests_network
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready" ]
      interval: 5s
      timeout: 3s
      retries: 10
  tests:
    container_name: tests
    build:
      context: .
      dockerfile: Dockerfile
    env_file:
      - ./.env
    environment:
      - DB_HOST=db_tests
    volumes:
      - ./src:/app/src/
    depends_on:
      db_tests:
        condition: service_healthy
    command: "pytest -vv"
    networks:
      - tests_network

networks:
  tests_network:
    name: tests_network

volumes:
  db_tests:
